lib/methods/clusterchecktask.cpp

```diff
@@ -58,19 +58,24 @@ void ClusterCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRe
 	std::pair<Dictionary::Ptr, Array::Ptr> feature_stats = CIB::GetFeatureStats();
 	cr->SetPerformanceData(feature_stats.second);
 
-	String connected_endpoints = FormatArray(status->Get("conn_endpoints"));
-	String not_connected_endpoints = FormatArray(status->Get("not_conn_endpoints"));
+	int numConnEndpoints = status->Get("num_conn_endpoints");
+	int numNotConnEndpoints = status->Get("num_not_conn_endpoints");
+
+	String output = "Icinga 2 Cluster";
+	cr->SetState(ServiceOK);
```

Comment:
This should be in the else branch (after line 74). Setting it to OK and then overwriting it with critical is a bit confusing.

Reply:
Mh ok, the compiler would optimize it anyway. I'll change it back.